design-library/src/components/BccPagination/BccPagination.vue

```diff
@@ -0,0 +1,160 @@
+import { toRefs, ref, computed, watchEffect } from "vue";
+import BccSelect from "../BccSelect/BccSelect.vue";
+
+type Props = {
+  items: any[];
+  rowsPerPageOptions?: number[];
+  rowsPerPage?: number;
+  maxButtonsDisplayed?: number;
+  displayLeftEllipsis?: boolean;
+  displayRightEllipsis?: boolean;
+  align?: "left" | "center" | "right";
+  displayRowsPerPage?: boolean;
+};
+
+type PageNumberOrEllipsis = number | "...";
+
+const props = withDefaults(defineProps<Props>(), {
+  displayLeftEllipsis: true,
+  displayRightEllipsis: true,
+  displayRowsPerPage: true,
+  maxButtonsDisplayed: 3,
+  align: "right",
+  rowsPerPage: 5,
+  rowsPerPageOptions: () => [5, 10, 25, 50],
+});
+const { items, maxButtonsDisplayed, rowsPerPage, displayRightEllipsis, displayLeftEllipsis } =
+  toRefs(props);
+
+const emit = defineEmits(["update:paginatedItems", "update:currentPage", "update:totalPages"]);
+
+const currentPage = ref(1);
+const perPage = ref(rowsPerPage.value.toString());
+const perPageNumber = computed(() => parseInt(perPage.value));
+const totalPages = computed(() => Math.ceil(items.value.length / perPageNumber.value));
+const maxButtons = computed(() => maxButtonsDisplayed.value);
+
+const isFirstPage = computed(() => currentPage.value === 1);
+const isLastPage = computed(() => currentPage.value === totalPages.value);
+
+const paginatedRows = computed(() => {
+  const start = (currentPage.value - 1) * perPageNumber.value;
+  const end = start + perPageNumber.value;
+  return items.value.slice(start, end);
+});
+
+const currentDisplayedPages = computed<PageNumberOrEllipsis[]>(() => {
+  let arr: PageNumberOrEllipsis[] = [];
+
+  let startPage = Math.max(1, currentPage.value - Math.floor(maxButtons.value / 2));
+  let endPage = Math.min(totalPages.value, currentPage.value + Math.floor(maxButtons.value / 2));
+
+  if (endPage - startPage < maxButtons.value) {
+    if (startPage === 1) {
+      endPage = Math.min(totalPages.value, startPage + maxButtons.value - 1);
+    } else if (endPage === totalPages.value) {
+      startPage = Math.max(1, endPage - maxButtons.value + 1);
+    }
+  }
+
+  if (startPage > 1 && displayLeftEllipsis.value === true) {
+    arr.push(1);
+    if (startPage > 2) {
+      arr.push("...");
+    }
+  }
+
+  for (let i: number = startPage; i <= endPage; i++) {
+    arr.push(i);
+  }
+
+  if (endPage < totalPages.value && displayRightEllipsis.value === true) {
+    if (endPage < totalPages.value - 1) {
+      arr.push("...");
+    }
+    arr.push(totalPages.value);
+  }
+
+  return arr;
+});
+
+const goToPage = (numPage: PageNumberOrEllipsis) => {
+  if (numPage === "...") {
+    return;
+  }
+
+  currentPage.value = numPage;
+};
+
+const changePage = (value: number) => {
+  currentPage.value += value;
+  emit("update:currentPage", currentPage.value);
+};
+
+watchEffect(() => {
+  emit("update:paginatedItems", paginatedRows.value);
+  emit("update:totalPages", totalPages.value);
+});
```

Comment:
Nice, I should use `watchEffect` more instead of plain `watch`, it works nicer for instances like this 👍